extensions/amp-base-carousel/0.1/child-layout-manager.js

```diff
@@ -405,7 +456,9 @@ export class ChildLayoutManager {
       return;
     }
 
+    this.underlyingAmpElementParent_ = {};
```

Comment:
When does `updateChildren` get called? Will the nearingViewportObserver_ callbacks still function properly after a carousel has its children updated?

Reply:
> When does updateChildren get called?

`updateChildren` gets called in the `buildCallback` of the parent elements (amp-base-carousel, amp-carousel, amp-stream-gallery). 

> Will the nearingViewportObserver_ callbacks still function properly after a carousel has its children updated?

 Yes, the observer should still function properly. When the buildCallback is run, the children should be updated. Then when the layoutCallback is run, the elements will be observed (creating a new mapping of SLIDE_IDs to slide elements).